sktime/benchmarking/benchmarking_dataclasses.py

```diff
@@ -0,0 +1,430 @@
+                results_json = json.load(f)
+            for row in results_json:
+                folds = {}
+                for fold_id, fold in row["folds"].items():
+                    scores = []
+                    for score in fold["scores"]:
+                        if isinstance(score["score"], dict):
+                            score_val = pd.DataFrame(score["score"])
+                        else:
+                            score_val = score["score"]
+                        scores.append(ScoreResult(score["name"], score_val))
+                    if "ground_truth" in fold:
+                        ground_truth = pd.Series(fold["ground_truth"])
+                    else:
+                        ground_truth = None
+                    if "predictions" in fold:
+                        predictions = pd.Series(fold["predictions"])
+                    else:
+                        predictions = None
+                    if "train_data" in fold:
+                        train_data = pd.Series(fold["train_data"])
+                    folds[int(fold_id)] = FoldResults(
+                        scores, ground_truth, predictions, train_data
+                    )
+
+                results.append(
+                    ResultObject(
+                        model_id=row["model_id"],
+                        task_id=row["task_id"],
+                        folds=folds,
+                    )
+                )
+            return results
+        except FileNotFoundError:
+            return []
+
+
+class ParquetStorageHandler(BaseStorageHandler):
+    """Storage handler for JSON files."""
+
+    # TODO this probably needs to be fixed!
+
+    def save(self, results: list[ResultObject]):
+        """Save the results to a JSON file.
+
+        Parameters
+        ----------
+        results : ResultObject
+            The results to save.
+        """
+        results_df = pd.json_normalize(
+            list(map(lambda x: asdict(x, pd_orient="tight"), results))
+        )
+
+        results_df = results_df.sort_values(by=["task_id", "model_id"])
+
+        # TODO store fails in hierachical case with level report
+        results_df = results_df.reset_index(drop=True)
+        results_df.to_parquet(self.path, index=False)
+
+    def load(self) -> list[ResultObject]:
+        """Load the results from a JSON file.
+
+        Returns
+        -------
+        ResultObject
+            The loaded results.
+
+        """
+        try:
+            results_df = pd.read_parquet(self.path)
+            results = []
+            for ix, row in results_df.iterrows():
+                folds = {}
+                for fold in row["folds"]:
+                    scores = []
+                    for score in fold["scores"]:
+                        scores.append(ScoreResult(score["name"], score["score"]))
+                    if "ground_truth" in fold:
+                        ground_truth = pd.Series(fold["ground_truth"])
```

Comment:
ground truth etc are not always `pd.Series`, e.g., in the multivariate case?

Reply:
Changed to DataFrame